CRM/ManualDirectDebit/Hook/Custom/StartDateGenerator.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Generates `Start Date` field if it wasn't filed by user
+ */
+class CRM_ManualDirectDebit_Hook_Custom_StartDateGenerator {
```

Comment:
I think CRM_ManualDirectDebit_Hook_Custom_MandateStartDateGenerator is a better name